src/utils/url.js

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+const defaultUrl = {
+  hash: "",
+  host: "",
+  hostname: "",
+  href: "",
+  origin: "null",
+  password: "",
+  pathname: "",
+  port: "",
+  protocol: "",
+  search: "",
+  searchParams: {}, // This should be a "URLSearchParams" object
+  username: ""
+};
+
+export function parse(url: string): URL | object {
+  try {
+    return new URL(url);
+  } catch (err) {
+    return defaultUrl;
```

Comment:
at this point, we should know we have URL

Reply:
Using `new URL(input)` is dangerous, as it can throw an error which would brick out our panel.  It's the same bug I fixed for NetMonitor.  Unless we have a URL validation process, I think this is a decent bet

Reply:
This also comes to mind:  https://github.com/devtools-html/debugger.html/issues/5102

Reply:
got it. then maybe we should do something like `new URL(<minimal viable url>)`